### PR TITLE
Set up ESLint and Prettier

### LINKED ISSUE #4: Add basic ESLint setup (For ESM) and Prettier
We should add a basic ESLint configuration (it can be based on https://github.com/nodejs/nodejs.org/blob/main/.eslintrc.json), except that it should only contain the rules for vanilla JavaScript (ESM) and nothing else (like Jest o React or TypeScript, etc...)

Same for Prettier, we should also add a Prettier configuration; we can also base it on the Node.js's Website ones (feel free to also grab the `.editorconfig`, and other base files).

We don't need Markdown or MDX linting/prettier rules as this repository itself won't hold any Markdown/MDX file.

### PR DESCRIPTION
**Pull request by araujogui**
Fixes #4

### CODE REVIEW

**Comment by cclauss** [comment]
As discussed at https://github.com/nodejs/api-docs-tooling/issues/4#issuecomment-2139921820 let's include a `pre-commit` or `husky` implementation that includes `codespell` so that identical tests can be run locally and on GitHub Actions.

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import pluginJs from '@eslint/js';
+import eslintConfigPrettier from 'eslint-config-prettier';
+import globals from 'globals';
+
+export default [
+  { languageOptions: { globals: globals.browser } },
+  pluginJs.configs.recommended,
```
Comment (AugustinMauroy) [inline eslint.config.mjs:7]:
```suggestion
  // See https://eslint.org/docs/latest/rules/ to understand theses rules
  pluginJs.configs.recommended,
```

What do you thinks about having link to docs ?

**Comment by araujogui** [comment]
> As discussed at [#4 (comment)](https://github.com/nodejs/api-docs-tooling/issues/4#issuecomment-2139921820) let's include a `pre-commit` or `husky` implementation that includes `codespell` so that identical tests can be run locally and on GitHub Actions.

I don't know how to install codespell here. I feel that we shouldn't use any Python dependency/package manager because it would increase the entry barriers for our future JavaScript contributors.

File: eslint.config.mjs
Comment (ovflowd) [inline eslint.config.mjs:1]: Is there any reason we're doing a JS-based config instead of a JSON one like on nodejs.org?

Comment (AugustinMauroy) [reply eslint.config.mjs:1]: it's just ESLint v9 ?
https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file

Comment (araujogui) [reply eslint.config.mjs:1]: > it's just ESLint v9 ? https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file

exactly

Comment (ovflowd) [reply eslint.config.mjs:1]: Right, but can't we use JSON on v9? It's just something myself and more people are familiar with I assume

Comment (araujogui) [reply eslint.config.mjs:1]: It's deprecated 
https://eslint.org/docs/latest/use/configure/configuration-files-deprecated

Comment (ovflowd) [reply eslint.config.mjs:1]: Oh

Comment (ovflowd) [reply eslint.config.mjs:1]: sadgers

Comment (bmuenzenmeyer) [reply eslint.config.mjs:1]: yea this is a large ecosystem shift we need to get used to 😶

**Review by ovflowd — APPROVED** [review]

File: eslint.config.mjs (hunk 1 above)
Comment (ovflowd) [reply eslint.config.mjs:7]: cc @araujogui

**Pull request merged**